File: src/Moryx.FactoryMonitor.Endpoints/Models/FactoryStateModel.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.FactoryMonitor.Endpoints.Models
{
    /// <summary>
    /// Simple model of factory
    /// </summary>
    [DataContract]
    public class FactoryStateModel : VisualizableItemModel
    {

        [DataMember]
        public virtual string BackgroundURL { get; set; }

        [DataMember]
        public virtual bool HasManufacturingFactory { get; set; }

        [DataMember]
        public List<OrderModel> OrderModels { get; set; }

        [DataMember]
        public List<ActivityChangedModel> ActivityChangedModels { get; set; }

        [DataMember]
        public List<CellStateChangedModel> CellStateChangedModels { get; set; }

        [DataMember]
        public List<ResourceChangedModel> ResourceChangedModels { get; set; }
    }
}

File: src/Moryx.FactoryMonitor.Web/src/app/api/models/Moryx/FactoryMonitor/Endpoints/Model/factory-state-model.ts
/* tslint:disable */
/* eslint-disable */
import { CellLocationModel as MoryxFactoryMonitorEndpointsModelCellLocationModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/cell-location-model';
import { CellModel as MoryxFactoryMonitorEndpointsModelCellModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/cell-model';
import { OrderModel as MoryxFactoryMonitorEndpointsModelOrderModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Model/order-model';
import { ActivityChangedModel as MoryxFactoryMonitorEndpointsModelsActivityChangedModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Models/activity-changed-model';
import { CellStateChangedModel as MoryxFactoryMonitorEndpointsModelsCellStateChangedModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Models/cell-state-changed-model';
import { ResourceChangedModel as MoryxFactoryMonitorEndpointsModelsResourceChangedModel } from '../../../../Moryx/FactoryMonitor/Endpoints/Models/resource-changed-model';
export interface FactoryStateModel {
  activityChangedModels?: null | Array<MoryxFactoryMonitorEndpointsModelsActivityChangedModel>;
  backgroundURL?: null | string;
  cellStateChangedModels?: null | Array<MoryxFactoryMonitorEndpointsModelsCellStateChangedModel>;
  /** @deprecated */cells?: null | Array<MoryxFactoryMonitorEndpointsModelCellModel>;
  hasManufacturingFactory?: boolean;
  iconName?: null | string;
  id?: number;
  isACell?: boolean;
  location?: MoryxFactoryMonitorEndpointsModelCellLocationModel;
  orderModels?: null | Array<MoryxFactoryMonitorEndpointsModelOrderModel>;
  resourceChangedModels?: null | Array<MoryxFactoryMonitorEndpointsModelsResourceChangedModel>;
}
